lib/chewy/strategy/atomic.rb

```diff
@@ -20,12 +20,20 @@ def update type, objects, options = {}
         ActiveSupport::Deprecation.warn("`urgent: true` option is deprecated and is not effective inside `:atomic` strategy, use `Chewy.strategy(:urgent)` strategy instead") if options.key?(:urgent)
 
         @stash[type] ||= []
-        @stash[type] |= type.adapter.identify(objects)
+        @stash[type] |= _identify_objects type, objects
       end
 
       def leave
         @stash.all? { |type, ids| type.import!(ids) }
       end
+
+      private
+
+      def _identify_objects type, objects
+        identifiers_from_evaluator = type.root_object.compose_id(objects) if type.root_object
```

Comment:
`compose_id` simply calls the proc passed to the root `:_id` option, so it simply calculates id for one object. But to check if those proc was passed - you have to use `if type.root_object.id` condition.

Objects is the thing you are passing from `update_index` block. It might be an object, an array or even a relation.